pkg/filter/dt/filter_mysql_prepare.go

```diff
@@ -74,18 +79,20 @@ func (f *_mysqlFilter) processBeforePrepareUpdate(ctx context.Context, conn *dri
 		return nil
 	}
 	executor := exec.NewPrepareUpdateExecutor(conn, updateStmt, stmt.BindVars, nil)
-	bi, err := executor.BeforeImage(ctx)
+	bi, err := executor.BeforeImage(newCtx)
 	if err != nil {
 		return err
 	}
-	if !proto.WithVariable(ctx, beforeImage, bi) {
+	if !proto.WithVariable(newCtx, beforeImage, bi) {
 		return errors.New("set before image failed")
 	}
 	return nil
 }
 
 func (f *_mysqlFilter) processAfterPrepareDelete(ctx context.Context, conn *driver.BackendConnection,
 	stmt *proto.Stmt, deleteStmt *ast.DeleteStmt) error {
+	newCtx, span := tracing.GetTraceSpan(ctx, "mysql_process_after_prepare_delete")
```

Comment:
I recommend not adding TraceContext for this location. instead, add inside of func `executor.BeforeImage`:
```
newCtx, span := tracing.GetTraceSpan(ctx, "fetch_before_image")
```